File: src/java/org/javaturk/dp/ch05/proxy/basbakan/BasbakanlikKalemi.java
package org.javaturk.dp.ch05.proxy.basbakan;

public class BasbakanlikKalemi {
	
	private Basbakan basbakan;
	
	public BasbakanlikKalemi(Basbakan basbakan){
		this.basbakan = new VekilBasbakan(basbakan);
	}

	public Basbakan banaBasbakaniVer(){
		System.out.println("Başbakanlık Kalemi: Tabi efendim");
		return basbakan;
	}
}

File: src/c#/ch01/BasbakanlikKalemi.cs
using System;
namespace DesignPatterns.org.javaturk.dp.basic.pattern.gof.structural.proxy.basbakan
{
    public class BasbakanlikKalemi
    {
        private IBasbakan basbakan;

        public BasbakanlikKalemi(IBasbakan basbakan)
        {
            this.basbakan = new VekilBasbakan(basbakan);
        }

        public IBasbakan BanaBasbakaniVer()
        {
            Console.WriteLine("Başbakanlık Kalemi: Tabi efendim");
            return basbakan;
        }
    }
}
